parquet-hadoop/src/main/java/org/apache/parquet/hadoop/DirectCodecFactory.java

```diff
@@ -68,7 +68,8 @@ class DirectCodecFactory extends CodecFactory implements AutoCloseable {
     try {
       tempClass = Class.forName("org.apache.hadoop.io.compress.DirectDecompressionCodec");
       tempCreateMethod = tempClass.getMethod("createDirectDecompressor");
-      tempDecompressMethod = tempClass.getMethod("decompress", ByteBuffer.class, ByteBuffer.class);
+      Class<?> tempClass2 = Class.forName("org.apache.hadoop.io.compress.DirectDecompressor");
+      tempDecompressMethod = tempClass2.getMethod("decompress", ByteBuffer.class, ByteBuffer.class);
```

Comment:
Yes. Because of that the class `FullDirectDecompressor` was never used but fell back to `IndirectDecompressor`. That's why we never have an actual failure because the functionality was covered.